iceoryx_posh/test/moduletests/test_roudi_config_json_file_provider.cpp

```diff
@@ -0,0 +1,442 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_posh/mepoo/segment_config.hpp"
+#include "iceoryx_posh/roudi/roudi_cmd_line_parser_config_file_option.hpp"
+#include "iceoryx_posh/roudi/roudi_config_json_file_provider.hpp"
+#include "test.hpp"
+#include <string>
+
+using namespace iox::roudi;
+using namespace iox::config;
+using namespace iox;
+using namespace ::testing;
+using namespace std;
+
+#if !defined(_WIN32)
+const std::string TestFile = "JSON_CONFIG.tmp";
+const std::string CrapFile = "CRAP_FILE.tmp";
+
+const std::string TempPath = "/tmp";
+const std::string TestFilePath = TempPath + "/" + TestFile;
+
+class JsonConfig_test : public Test
+{
+  public:
+    JsonConfig_test()
+        : Test()
+    {
+    }
+    virtual void SetUp()
+    {
+        char str[] = "{\n"
+                     "\t\"general\":{\n"
+                     "\t\t\"version\" : 1\n"
+                     "\t},"
+                     "\t\"segment\":[\n"
+                     "\t\t{\n"
+                     "\t\t\t\"writer\":\"foo\",\n"
+                     "\t\t\t\"reader\":\"bar\",\n"
+                     "\t\t\t\"mempool\":[\n"
+                     "\t\t\t\t{\n"
+                     "\t\t\t\t\t\"size\":32,\n"
+                     "\t\t\t\t\t\"count\":10000\n"
+                     "\t\t\t\t},\n"
+                     "\t\t\t\t{\n"
+                     "\t\t\t\t\t\"size\":128,\n"
+                     "\t\t\t\t\t\"count\":10000\n"
+                     "\t\t\t\t}\n"
+                     "\t\t\t]\n"
+                     "\t\t}\n"
+                     "\t]\n"
+                     "}\n";
+
+        ofstream tmpFile;
+        tmpFile.open(TestFilePath.c_str());
+        tmpFile << str;
+        tmpFile.close();
+    }
+
+    virtual void TearDown()
+    {
+        optind = 0;
+    }
+};
+
+class JsonConfig_Failure_test : public Test
+{
+  public:
+    JsonConfig_Failure_test()
+        : Test()
+    {
+    }
+    virtual void SetUp()
+    {
+    }
+    virtual void TearDown()
+    {
+        optind = 0;
+    }
+
+    iox::cxx::expected<iox::RouDiConfig_t, iox::roudi::RouDiConfigFileParseError> parseJson(std::string json)
+    {
+        iox::config::CmdLineParserConfigFileOption cmdLineParser;
+        int argc = 3;
+        std::string arg1("executable");
+        std::string arg2("--config-file");
+        std::string arg3("/tmp/JSON_CONFIG.tmp");
+        char* argv[3] = {&arg1[0], &arg2[0], &arg3[0]};
+
+        ofstream tmpFile;
+        tmpFile.open(TestFilePath.c_str());
+        tmpFile << json;
+        tmpFile.close();
+
+        auto cmdLineArgs = cmdLineParser.parse(argc, argv);
+
+        JsonRouDiConfigFileProvider reader(cmdLineArgs.value());
+        return reader.parse();
+    }
+};
+
+TEST_F(JsonConfig_test, TestReader)
```

Comment:
I think there are some tests missing. Please read our guide: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

And then add tests with a mempool config of zero, one, many, limit (MAX_NUMBERS_OF_MEMPOOLS) and limit + 1.

I think the same goes for segment.


Additionally we require tests for entries with a negative count or size of mempools or where the count/size exceeds the storage capacity of an `uint32_t`.